pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -400,20 +524,35 @@ public long getPartitionEndToEndIngestionDelayMs(int partitionId) {
   }
 
   public long getPartitionIngestionOffsetLag(int partitionId) {
-    IngestionInfo ingestionInfo = _ingestionInfoMap.get(partitionId);
-    if (ingestionInfo == null) {
-      return 0;
-    }
-    StreamPartitionMsgOffset currentOffset = ingestionInfo._currentOffset;
-    StreamPartitionMsgOffset latestOffset = ingestionInfo._latestOffset;
-    if (currentOffset == null || latestOffset == null) {
-      return 0;
-    }
-    // TODO: Support other types of offsets
-    if (!(currentOffset instanceof LongMsgOffset && latestOffset instanceof LongMsgOffset)) {
+    try {
+      IngestionInfo ingestionInfo = _ingestionInfoMap.get(partitionId);
+      if (ingestionInfo == null) {
+        return 0;
+      }
+      StreamPartitionMsgOffset currentOffset = ingestionInfo._currentOffset;
+      StreamPartitionMsgOffset latestOffset = ingestionInfo._latestOffset;
```

Comment:
(MAJOR) This is almost never accurate because current offset and latest offset are updated separately. We probably should directly track the lag and update the lag when fetching the latest offset.
Seems `currentOffset` is used only here, so we can just remove both `offsets` and only keep `lag` in `IngestionInfo`

Reply:
That's the issue, that we can get the currentOffset only via the updateIngestionMetric method call
and the latest offset we are calculating in seperate periodic thread

if I just want to store the lag, i'll either have to calculate latestOffset every time `updateIngestionMetrics` is called with currentOffset for the partition. I can't do this since it'll defeat the purpose of this PR.